components/net/image_cache.rs

```diff
@@ -405,14 +401,39 @@ impl ImageCacheImpl {
     }
 }
 
+struct ImageDecoderRunnable {
+    store: Arc<Mutex<ImageCacheStore>>,
+    key: PendingImageId,
+    bytes: Arc<Vec<u8>>,
+}
+
+impl ImageDecoderRunnable {
+    fn run(&self) {
+        let local_store = self.store.clone();
+        let bytes = self.bytes.clone();
+        let key = self.key.clone();
+        thread::spawn(move || {
+            let msg = decode_bytes_sync(key, &*bytes);
+            debug!("Image decoded");
+            local_store.lock().unwrap().handle_decoder(msg);
+        });
+    }
+}
+
+pub struct ImageCacheImpl {
```

Comment:
@jdm if I am not wrong, this means that I need to use `net::image_cache::ImageCacheHandle` instead of `net_traits::image_cache::ImageCache` in all consumers, is this ok?